litebird_sim/distribute.py

```diff
@@ -0,0 +1,215 @@
+# -*- encoding: utf-8 -*-
+
+from collections import namedtuple
+
+Span = namedtuple("Span", ["start_idx", "num_of_elements"])
+"""A sub-range in a sequence of elements.
+
+It has two fields: `start_idx` and `num_of_elements`.
+
+"""
+
+
+def distribute_evenly(num_of_elements, num_of_groups):
+    """Evenly distribute a set of equal elements between groups.
+
+    Assuming that we have `num_of_elements` objects that we want to
+    assign to `num_of_groups` separate groups, this function proposes
+    a distribution that tries to assign the most uniform number of
+    elements to each group.
+
+    This function works even if ``num_of_elements < num_of_groups``.
+
+    .. doctest::
+
+      >>> distribute_evenly(5, 2)
+      [Span(start_idx=0, num_of_elements=3), Span(start_idx=3, num_of_elements=2)]
+
+      >>> distribute_evenly(1, 2)
+      [Span(start_idx=0, num_of_elements=1), Span(start_idx=1, num_of_elements=0)]
+
+    Args:
+        num_of_elements (int): The number of elements to distribute
+        num_of_groups (int): The number of groups to use in the result
+
+    Returns:
+        a list of 2-element tuples containing `num_of_groups`
+        elements, each of them being a 2-element tuple containing (1)
+        the index of the element in the group and (2) the number of
+        elements in the group. Being named tuples, you can access the
+        index using the field name ``start_idx``, and the number of
+        elements using the name ``num_of_elements``.
+
+    """
+    assert num_of_elements > 0
+    assert num_of_groups > 0
+
```

Comment:
It may look obscure, but what about
```
cur_pos = 0
res = []
for i in range(num_of_groups): 
    cur_length = (num_of_elements - (i + 1)) // num_of_groups + 1
    result.append(Span(start_idx=cur_pos, num_of_elements=cur_length))
    cur_pos += cur_length
````

Reply:
Is this to avoid the `if` within the `for` loop? Do I understand correctly?

Reply:
Yes, the if and also this mechanism of the leftovers. But don't bother for now, let's discuss on the patch I'll propose maybe later today